android/tutorials/tutorial3/README.md

```diff
@@ -275,12 +275,29 @@ Building the project at this point, Dagger should give you:
 
 ## Bubbling up dependencies
 
-This will be super easy, if you got this far. The only difference is that if we cannot provide a dependency directly, we also add it to the `Dependency` interface of the parent (and then to its parent and so on until we can actually provide id).
+This will be super easy, if you got this far. The only difference is that if we cannot provide a dependency directly. We also add it to the `Dependency` interface of our rib and to the 'Dependency' interfaces of all parent ribs until this rib can actually provide id.
 
 Really, just a one-liner.
 
 Try it:
 
+```kotlin
+interface GreetingsContainer : Rib {
```

Comment:
The code snippet for `GreetingsContainer` is literally below this one, and is the correct one. It's only `User` dependency that needs to bubble up, `Config` is provided directly by `GreetingsContainer` in `GreetingsContainerModule`.

Reply:
I meant `HelloWorld` class, but for documentation I copied `GreetingsContainer` and forgot to rename the whole class, I'll fix it in a moment